packages/vm/src/imports.rs

```diff
@@ -61,9 +71,20 @@ pub fn do_write<S: Storage, Q: Querier>(
 
     let key = read_region(ctx, key_ptr, MAX_LENGTH_DB_KEY)?;
     let value = read_region(ctx, value_ptr, MAX_LENGTH_DB_VALUE)?;
-    let (_, gas_info) =
-        with_storage_from_context::<S, Q, _, _>(ctx, |store| Ok(store.set(&key, &value)?))?;
-    process_gas_info::<S, Q>(ctx, gas_info)?;
+
+    let ffi_result =
+        with_storage_from_context::<S, Q, _, _>(ctx, |store| Ok(store.set(&key, &value)))?;
+
+    let _ = match ffi_result {
+        Ok(((), gas_info)) => {
+            process_gas_info::<S, Q>(ctx, gas_info)?;
+            Ok(())
+        }
+        Err((err, gas_info)) => {
+            process_gas_info::<S, Q>(ctx, gas_info)?;
+            Err(VmError::from(err))
+        }
+    }?;
```

Comment:
you don't need the `let _ = ` in these, since the value of the expression after the `?` is just `()`. You usually use `let _ = ` for expressions whose type is `#[must_use]` such as `Result`s

Reply:
Returning from the match expression feels strange without assigning to something. But I agree it is correct and sufficient.

Reply:
Well... You could remove the `Ok(())` line and the `?` from the end, and replace `Err(VmError::from(err))` with `Err(VmError::from(err))?;`

Reply:
Done in https://github.com/CosmWasm/cosmwasm/pull/486/commits/b84f775912ae80e825e87649fc76aab11b484134. I think it helps when all `match ffi_result` blocks in this file have almost the same structure, no matter if there is a value or `()` returned.

Reply:
haha, writing `let _ = ...` was hiding a bug (missing error propagation) in the other place it was used

Reply:
:scream: